models/issues/issue_list.go

```diff
@@ -599,3 +599,55 @@ func (issues IssueList) GetApprovalCounts(ctx context.Context) (map[int64][]*Rev
 
 	return approvalCountMap, nil
 }
+
+func (issues IssueList) BlockingDependenciesMap(ctx context.Context) (issueDepsMap map[int64][]*DependencyInfo, err error) {
+	var issueDeps []*DependencyInfo
+
+	err = db.GetEngine(ctx).
+		Table("issue").
+		Join("INNER", "repository", "repository.id = issue.repo_id").
```

Comment:
The join of repository seems unnecessary.

Reply:
The tests requires this
So do I need to change the test or do I remove this line?

Reply:
Did you mean `TestIssueList_BlockingDependenciesMap`? Maybe we can check `issue.RepoID` and remove the join and remove the field `Repository` of `DependencyInfo`?

Reply:
Yes `TestIssueList_BlockingDependenciesMap` needs the repo.
The older functions which uses `DependencyInfo` uses repo also. So I cant remove repo from `DependencyInfo` but I can add a new struct and use the new one,  then I can remove the join.